### ISSUE TITLE
No coverage report when only targeting .NET 5.0

### ISSUE THREAD

**Issue by AdmiringWorm** [cake-contrib/Cake.Recipe#797]
When using Cake.Recipe and only targeting .NET 5.0, there are no coverage reports generated (nor are coverlet ran correctly).

This seems to be that we are checking whether the Project File is reported as a .NET Core project (using Cake.Incubator), and in the case of only targeting .NET 5.0 this returns false.
We should run the same code as we run on .NET Core when targeting .NET 5.0.

**Comment by AdmiringWorm** [comment]
From the looks of it, this also causes .NET Publish not to run on .NET 5.0 frameworks.

**Comment by nils-a** [comment]
#863 is actually a duplicate of this.

See also https://github.com/cake-contrib/Cake.Incubator/issues/116.

**Comment by nils-a** [comment]
Updating Cake.Incubator to 7.0.0 will fix this. However, that update is blocked by #872

**Comment by AdmiringWorm** [comment]
I'll be closing this one, as it is expected to be solved by improving how coverlet is being run.
This work is covered by #570
